netsync/manager.go

```diff
@@ -847,22 +823,136 @@ func (sm *SyncManager) handleBlockMsg(bmsg *blockMsg) {
 		}
 	}
 
+	return isCheckpointBlock, nil
+}
+
+// handleBlockMsg handles block messages from all peers.
+func (sm *SyncManager) handleBlockMsg(bmsg *blockMsg) {
+	peer := bmsg.peer
+	state, exists := sm.peerStates[peer]
+	if !exists {
+		log.Warnf("Received block message from unknown peer %s", peer)
+		return
+	}
+
+	// If we didn't ask for this block then the peer is misbehaving.
+	blockHash := bmsg.block.Hash()
+	if _, exists := state.requestedBlocks[*blockHash]; !exists {
+		// The regression test intentionally sends some blocks twice
+		// to test duplicate block insertion fails.  Don't disconnect
+		// the peer or ignore the block when we're in regression test
+		// mode in this case so the chain code is actually fed the
+		// duplicate blocks.
+		if sm.chainParams != &chaincfg.RegressionNetParams {
```

Comment:
In this case I think it's the tests should be updated, not the code, otherwise we cannot cover the branch here in our tests.

Reply:
Ok so this code doesn't exist anymore but it's something that's already existed in the code so I think it's out of the scope for this PR.